service/tx/ons.go

```diff
@@ -0,0 +1,159 @@
+		Price:   args.Price,
+	}
+
+	uuidNew, _ := uuid.NewUUID()
+	fee := action.Fee{args.Fee, args.Gas}
+	tx := &action.BaseTx{
+		Data: domainCreate,
+		Fee:  fee,
+		Memo: uuidNew.String(),
+	}
+
+	packet, err := serialize.GetSerializer(serialize.NETWORK).Serialize(tx)
+	if err != nil {
+		return errors.Wrap(err, "err while network serialization")
+	}
+
+	*reply = client.SendTxReply{
+		RawTx: packet,
+	}
+
+	return nil
+}
+
+func (s *Service) Ons_CreateRawUpdate(args client.OnsUpdateRequest, reply *client.SendTxReply) error {
+
+	domainUpdate := ons.DomainUpdate{
+		Owner:   keys.Address(args.Owner),
+		Account: keys.Address(args.Account),
+		Name:    args.Name,
+		Active:  args.Active,
+	}
+
+	uuidNew, _ := uuid.NewUUID()
+	fee := action.Fee{args.Fee, args.Gas}
+	tx := &action.BaseTx{
+		Data: domainUpdate,
+		Fee:  fee,
+		Memo: uuidNew.String(),
+	}
+
+	packet, err := serialize.GetSerializer(serialize.NETWORK).Serialize(tx)
+	if err != nil {
+		return errors.Wrap(err, "err while network serialization")
+	}
+
+	*reply = client.SendTxReply{
+		RawTx: packet,
+	}
+
+	return nil
+}
+
+func (s *Service) Ons_CreateRawSale(args client.OnsSaleRequest, reply *client.SendTxReply) error {
+
+	domainSale := ons.DomainSale{
+		DomainName:   args.Name,
+		OwnerAddress: keys.Address(args.OwnerAddress),
+		Price:        args.Price,
+		CancelSale:   args.CancelSale,
+	}
+
+	uuidNew, _ := uuid.NewUUID()
+	fee := action.Fee{args.Fee, args.Gas}
+	tx := &action.BaseTx{
+		Data: domainSale,
+		Fee:  fee,
+		Memo: uuidNew.String(),
+	}
+
+	packet, err := serialize.GetSerializer(serialize.NETWORK).Serialize(tx)
+	if err != nil {
+		return errors.Wrap(err, "err while network serialization")
+	}
+
+	*reply = client.SendTxReply{
+		RawTx: packet,
+	}
+
+	return nil
+}
+
+func (s *Service) Ons_CreateRawBuy(args client.OnsPurchaseRequest, reply *client.SendTxReply) error {
+
+	domainPurchase := ons.DomainPurchase{
+		Name:     args.Name,
+		Buyer:    keys.Address(args.Buyer),
+		Account:  keys.Address(args.Account),
+		Offering: args.Offering,
+	}
+
+	uuidNew, _ := uuid.NewUUID()
+	fee := action.Fee{args.Fee, args.Gas}
+	tx := &action.BaseTx{
+		Data: domainPurchase,
+		Fee:  fee,
+		Memo: uuidNew.String(),
+	}
+
+	packet, err := serialize.GetSerializer(serialize.NETWORK).Serialize(tx)
+	if err != nil {
+		return errors.Wrap(err, "err while network serialization")
+	}
+
+	*reply = client.SendTxReply{
+		RawTx: packet,
+	}
+
+	return nil
+}
+
+func (s *Service) Ons_CreateRawSend(args client.OnsSendRequest, reply *client.SendTxReply) error {
+
+	domainSend := ons.DomainSend{
+		DomainName: args.Name,
+		From:       args.From,
+		Amount:     args.Amount,
+	}
+
+	uuidNew, _ := uuid.NewUUID()
+	fee := action.Fee{args.Fee, args.Gas}
+	tx := &action.BaseTx{
+		Data: domainSend,
+		Fee:  fee,
+		Memo: uuidNew.String(),
+	}
+
+	packet, err := serialize.GetSerializer(serialize.NETWORK).Serialize(tx)
+	if err != nil {
+		return errors.Wrap(err, "err while network serialization")
```

Comment:
When returning rpc errors use `rpc.NewError` or one of the `rpc.Code` related errors. I'd use `rpc.InternalError` in this case

Reply:
Do this for all errors you return